autopush/base.py

```diff
@@ -77,3 +93,35 @@ def authenticate_peer_cert(self):
         self.set_header('WWW-Authenticate',
                         'Transport mode="tls-client-certificate"')
         self.finish()
+
+    def _write_response(self, status_code, errno=None, message=None,
+                        error=None, headers=None, url=DEFAULT_ERR_URL):
+        """Writes out a full JSON error and sets the appropriate status"""
+        self.set_status(status_code, reason=error)
+        error_data = dict(
+            code=status_code,
+            error=error or status_codes.get(status_code, ""),
+            more_info=url,
+        )
+        if errno:
+            error_data["errno"] = errno
+        if message:
+            error_data["message"] = message
+        self.write(json.dumps(error_data))
+        self.set_header("Content-Type", "application/json")
+        if headers:
+            for header in headers.keys():
+                self.set_header(header, headers.get(header))
+        self.finish()
+
+
+class DefaultHandler(BaseHandler):
+    """Unauthenticated catch-all handler that returns a 404 for
+    unknown paths. Cyclone matches handlers in order, so this handler
+    should be registered last."""
+
+    def authenticate_peer_cert(self):
```

Comment:
this authenticate_peer_cert override isn't needed here.

In fact it's not necessary for DefaultHandler to subclass BaseHandler at all (just don't bother passing it h_kwargs in main) but I'm ok w/ keeping it a BaseHandler subclass.